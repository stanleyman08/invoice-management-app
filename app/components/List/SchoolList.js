import React from 'react';
import PropTypes from 'prop-types';

import MaterialTable from 'material-table';
import Search from '@material-ui/icons/Search';
import ResetSearch from '@material-ui/icons/Clear';
import FirstPage from '@material-ui/icons/FirstPage';
import LastPage from '@material-ui/icons/LastPage';
import NextPage from '@material-ui/icons/ChevronRight';
import PreviousPage from '@material-ui/icons/ChevronLeft';
import Delete from '@material-ui/icons/DeleteOutline';
import Edit from '@material-ui/icons/Edit';

const SchoolList = ({
  schools,
  deleteSchool,
  onHandleEditClickOpen,
  history
}) => (
  <div style={{ maxWidth: '100%' }}>
    <MaterialTable
      columns={[{ title: 'Name', field: 'name' }]}
      data={schools}
      actions={[
        {
          icon: Edit,
          onClick: (event, rowData) => onHandleEditClickOpen(rowData.name)
        },
        {
          icon: Delete,
          onClick: (event, rowData) => deleteSchool(rowData._id)
        }
      ]}
      onRowClick={(event, selectedRow) =>
        history.push(`/app/orders/${selectedRow._id}`)
      }
      options={{
        showTitle: false,
        actionsColumnIndex: -1
      }}
      icons={{
        Search,
        ResetSearch,
        FirstPage,
        LastPage,
        NextPage,
        PreviousPage
      }}
    />
  </div>
);

SchoolList.propTypes = {
  schools: PropTypes.arrayOf(Object).isRequired,
  deleteSchool: PropTypes.func.isRequired,
  onHandleEditClickOpen: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired
};

export default SchoolList;
