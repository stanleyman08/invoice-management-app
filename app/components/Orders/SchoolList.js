import React from "react";
// @material-ui/core components
import withStyles from "@material-ui/core/styles/withStyles";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import IconButton from '@material-ui/core/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';

import { Link } from "react-router-dom";

import { deleteSchool } from "../../utils/SchoolAPI.js"

const styles = {
    removeUnderline: {
        textDecoration: "none"
    }
}

class SchoolList extends React.Component {
    constructor(props) {
        super(props);
    };

    handleDelete = (id) => {
        deleteSchool(id).then(() => {
        	this.props.onLoadData();
        });
    };
    render() {
        const { classes } = this.props;
        const listSchools = this.props.schoolsData.map((school, i) =>
            <div key={i}>
                <ListItem button component={Link} to={`/orders/${i}`}>
                    <ListItemText primary={school.name}/>
                    <ListItemSecondaryAction onClick={() => this.handleDelete(school["_id"])}>
                        <IconButton aria-label="Delete" >
                            <DeleteIcon />
                        </IconButton>
                    </ListItemSecondaryAction>
                </ListItem>
            </div>
        );
        return (
            <List>
                {listSchools}
            </List>
        );
    };
}

export default withStyles(styles)(SchoolList);