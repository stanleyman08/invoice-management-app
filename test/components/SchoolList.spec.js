import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router';
import SchoolList from '../../app/components/List/SchoolList.js';

// jest.mock('../../app/utils/SchoolAPI.js', () => ({
//   deleteSchool: jest.fn()
// }));

Enzyme.configure({ adapter: new Adapter() });

describe('SchoolList component', () => {
  it('renders without throwing an error', () => {
    const schools = [];
    const component = shallow(<SchoolList schools={schools} />);
    expect(component).toMatchSnapshot();
  });

  it('shows no schools if schools is empty', () => {
    const schools = [];
    const component = mount(
      <MemoryRouter>
        <SchoolList schools={schools} />
      </MemoryRouter>
    );
    expect(component.find('ListItem')).toHaveLength(0);
  });

  it('shows a list of schools if schools exist: 1 school', () => {
    const schools = [{ name: 'Churchill', orders: [], _id: 'mockid1' }];
    const component = mount(
      <MemoryRouter>
        <SchoolList schools={schools} />
      </MemoryRouter>
    );
    expect(component.find('ListItem')).toHaveLength(1);
    expect(component.find('ListItem').text()).toEqual('Churchill');
  });

  it('shows a list of schools if schools exist: multiple schools', () => {
    const schools = [
      { name: 'Churchill', orders: [], _id: 'mockid1' },
      { name: 'Windermere', orders: [], _id: 'mockid2' },
      { name: 'Van Tech', orders: [], _id: 'mockid3' }
    ];
    const component = mount(
      <MemoryRouter>
        <SchoolList schools={schools} />
      </MemoryRouter>
    );
    expect(component.find('ListItem')).toHaveLength(3);
  });
});
