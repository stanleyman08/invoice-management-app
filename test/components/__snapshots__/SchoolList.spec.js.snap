// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchoolList component renders without throwing an error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SchoolList
    schools={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Component)
        actions={
          Array [
            Object {
              "icon": [Function],
              "onClick": [Function],
            },
            Object {
              "icon": [Function],
              "onClick": [Function],
            },
          ]
        }
        columns={
          Array [
            Object {
              "field": "name",
              "title": "Name",
            },
          ]
        }
        data={Array []}
        icons={
          Object {
            "FirstPage": [Function],
            "LastPage": [Function],
            "NextPage": [Function],
            "PreviousPage": [Function],
            "ResetSearch": [Function],
            "Search": [Function],
          }
        }
        onRowClick={[Function]}
        options={
          Object {
            "actionsColumnIndex": -1,
            "showTitle": false,
          }
        }
      />,
      "style": Object {
        "maxWidth": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "actions": Array [
          Object {
            "icon": [Function],
            "onClick": [Function],
          },
          Object {
            "icon": [Function],
            "onClick": [Function],
          },
        ],
        "columns": Array [
          Object {
            "field": "name",
            "title": "Name",
          },
        ],
        "data": Array [],
        "icons": Object {
          "FirstPage": [Function],
          "LastPage": [Function],
          "NextPage": [Function],
          "PreviousPage": [Function],
          "ResetSearch": [Function],
          "Search": [Function],
        },
        "onRowClick": [Function],
        "options": Object {
          "actionsColumnIndex": -1,
          "showTitle": false,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Component)
          actions={
            Array [
              Object {
                "icon": [Function],
                "onClick": [Function],
              },
              Object {
                "icon": [Function],
                "onClick": [Function],
              },
            ]
          }
          columns={
            Array [
              Object {
                "field": "name",
                "title": "Name",
              },
            ]
          }
          data={Array []}
          icons={
            Object {
              "FirstPage": [Function],
              "LastPage": [Function],
              "NextPage": [Function],
              "PreviousPage": [Function],
              "ResetSearch": [Function],
              "Search": [Function],
            }
          }
          onRowClick={[Function]}
          options={
            Object {
              "actionsColumnIndex": -1,
              "showTitle": false,
            }
          }
        />,
        "style": Object {
          "maxWidth": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "actions": Array [
            Object {
              "icon": [Function],
              "onClick": [Function],
            },
            Object {
              "icon": [Function],
              "onClick": [Function],
            },
          ],
          "columns": Array [
            Object {
              "field": "name",
              "title": "Name",
            },
          ],
          "data": Array [],
          "icons": Object {
            "FirstPage": [Function],
            "LastPage": [Function],
            "NextPage": [Function],
            "PreviousPage": [Function],
            "ResetSearch": [Function],
            "Search": [Function],
          },
          "onRowClick": [Function],
          "options": Object {
            "actionsColumnIndex": -1,
            "showTitle": false,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
